tests/unittests/partitionTest.cpp

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "BackendTestUtils.h"
+
+#include "glow/Graph/Graph.h"
+#include "glow/Optimizer/Partition.h"
+
+#include "gtest/gtest.h"
+
+using namespace glow;
+
+class PartitionTest : public ::testing::Test {
+public:
+  PartitionTest() : F_(mod_.createFunction("main")) {}
+
+protected:
+  Module mod_;
+  Function *F_;
+};
+
+TEST_F(PartitionTest, SerialExecution) {
+  auto *input =
+      mod_.createVariable(ElemKind::FloatTy, {1, 32}, "input",
+                          VisibilityKind::Public, Variable::TrainKind::None);
+
+  // Initial FC.
+  Node *I = F_->createFullyConnected("initial_fc", input, 16);
+  I = F_->createSigmoid("initial_sigmoid", I);
+
+  // Left branch.
+  Node *L = F_->createFullyConnected("left_fc1", I, 16);
+  L = F_->createSigmoid("left_sigmoid1", L);
+  L = F_->createFullyConnected("left_fc2", L, 8);
+  L = F_->createSigmoid("left_sigmoid2", L);
+
+  // Right branch.
+  Node *R = F_->createFullyConnected("right_fc1", I, 16);
+  R = F_->createSigmoid("right_sigmoid1", R);
+  R = F_->createFullyConnected("right_fc2", R, 8);
+  R = F_->createSigmoid("right_sigmoid2", R);
+
+  // Join branches.
+  auto *mul = F_->createMul("mul", L, R);
+  auto *save = F_->createSave("ret", mul);
+  auto &res = save->getVariable()->getPayload();
+
+  auto G = glow::partition(F_);
+  ASSERT_EQ(G.getFunctions().size(), 4);
+
+  auto it = G.getFunctions().begin();
+  {
+    auto *F = *it++;
+    EXPECT_EQ(F->getNodes().size(), 3);
```

Comment:
In your tests, you check the number of nodes in the partition and the number of inputs. Do we need to check also for some structural properties? E.g. that certain variables or Save nodes are present in the partition?

Reply:
Might be reasonable.  In development I found the simple structural checks plus the execution check caught more or less everything and are pretty simple to look at :)